language-design.md

```diff
@@ -16,6 +16,22 @@ are:
 For each quote, describe what it was about the quote that led you pick it.
 
 **Response**
+# Quote 1, Mernik et all 2002, Section 2.2
+>"As mentioned in Section 1.1, a quantitative treatment of the trade-offs involved is difficult. In practice, short-term consid- erations and lack of expertise may easily cause indefinite postponement of the decision." [Mernik et all 2002 section 2.2]
+
+This quote really resonated with us because we realized that doing what is comfortable often leads to not doing what is hard (or has high startup costs) which would potentially deter developers from building a new language to develop in, even if that language might be more comfortable in the long run. We also decided that even though building/designing a DSL is not _that_ hard it does require mental effort almost every step of the way. First you need to decide that its necessary, then you need to make implementation choices etc. When building a large project its sometimes difficult to look past short term considerations (like getting code written) in order to see long term benefits of potentially building a DSL. We also noticed that this quote makes for good life advice: don't postpone hard things because eventually postponing just means not doing. 
+
+# Quote 2, Bloch 2006
+>"If it’s hard to find good names, go back to the drawing board.
+Don’t be afraid to split or merge an API, or embed it in a more general setting. If names start falling into place, you’re on the right track." [Bloch 2006]
+
+We chose this quote because we agreed with it in some contexts but also thought that it didn't necessarily apply to all cases, situations or project types. We thought that this quote was good advice for building an API (when other people would be using it) but we weren't sure whether it applied to programming larger scale project. It definitely is a really intersting gauge to use and an easy and illuminating question to ask yourself as a devloper in order to get a read on how clear your thoughts are about what you're building and why. The problem we had with it was that we weren't convinced that every useful/necessary thing in a project would have an obvious name. What about helper functions? Those often serve specific and important purposes but don't always have obvious names. Consider the example of binary tree traversal--this relatively simple concept requires helper functions that dont necessarily have obvious names. 
+
```

Comment:
Yeah, I haven't had experience with this technique, so I'm not sure that it is useful for general programming. It does make sense that every function should have a very clear purpose, and honestly, if a function can't be described in just a few words, then you should split it into multiple small functions.